final_task/rss_reader/functions/check_func.py

```diff
@@ -0,0 +1,54 @@
+"""Contain checking functions"""
+
+import requests
+import logging
+import sys
+import classes.exceptions as exc
+import coloredlogs
+
+def check_internet_connection(logger):
+    """Check if network connection is avaliable"""
+    try:
+        logger.info("Checking internet connection...")
+        requests.get('https://www.google.com/', timeout=1)
+        return True
+    except requests.ConnectionError:
+        raise exc.InternetConnectionError("No internet access")
+
+
+def check_verbose(command_line_args):
+    """If vebose argument is True, set logging to info level"""
+    if command_line_args.colorize:
+        coloredlogs.install()
+    logging.basicConfig(format='[%(asctime)s][%(levelname)s] %(message)s',
+                        stream=sys.stdout,
+                        level=logging.INFO
+                        )
+    logger = logging.getLogger("rss_reader_logger")
+    if command_line_args.verbose:
+        logger.setLevel(level=logging.INFO)
+    else:
+        logger.setLevel(level=logging.WARNING)
+    return logger
+
+
+def check_feed_status(feed):
+    """Check HTTP status code"""
+    try:
+        status = feed.status
+        if 500 > status >= 400:
+            raise exc.GettingFeedError("Client error – the request contains bad syntax or cannot be fulfilled.")
+        elif 600 > status >= 500:
+            raise exc.GettingFeedError("Server error – the server failed to fulfill an apparently valid request.")
+        else:
+            return feed
```

Comment:
It is better to use `bozo` attribute for this check.

`feed` here has type `FeedParserDict`, which is inherited from `dict`, so it is a dictionary, which has keys and values.
Neither `FeedParserDict` nor `dict` have attribute `status`, so theoretically you should always get an `AttributeError` here.
BUT `FeedParserDict` class has its magic method `__getattribute__` redefined (note: getattribute is called every time you try to access some object attribute using `.`)  in a way so if there is no such attribute (for example `status`), it will try to look in this dictionary keys, using `__getitem__` (note: getitem is called when you try to acccess some item in collection by key or by index: `data["status"]`). So there is no attribute `status`, but `status` is a key in `data` dictionary, so it is still luckily works.

In short, using `status` here is pretty volatile and may break your program.